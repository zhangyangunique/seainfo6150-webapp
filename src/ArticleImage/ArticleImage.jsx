import React from "react";
import PropTypes from "prop-types";
import styles from "./ArticleImage.module.css";

const ArticleImage = (props) => {
	return (
		<img className={styles.image} src={props.url} alt={props.title} />
	);
};

ArticleImage.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
};
export default ArticleImage; 
