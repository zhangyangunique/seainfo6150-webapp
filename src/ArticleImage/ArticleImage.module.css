.image {
	border-radius: 50%;
	border: 1rem solid #c00;
	box-shadow: 0px 5px 10px 5px rgba(0, 0, 0, 0.5);
	margin-bottom: 30px;
	width: 12rem;
	height: 9rem;
}

@media screen and (min-width: 525px) {
	.image {
		width: 11rem;
		height: 8rem;
	}
}

@media screen and (min-width: 800px) {
	.image {
		width: 10rem;
		height: 7rem;
	}
}

@media screen and (min-width: 1150px) {
	.image {
		width: 9rem;
		height: 6rem;
	}
}