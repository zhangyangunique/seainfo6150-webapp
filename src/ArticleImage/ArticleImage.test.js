import React from "react";
import ArticleImage from "./ArticleImage";

const title = "The Statue of Liberty's torch heads to new museum";
const url = "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg";

describe("ArticleImage tests", () => {
	it("renders correctly", () => {
		const { container } = render(<ArticleImage title={title} url={url} />);
		expect(container).toMatchSnapshot();
	});
});