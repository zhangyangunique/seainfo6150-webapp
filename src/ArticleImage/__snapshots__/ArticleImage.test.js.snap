// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleImage tests renders correctly 1`] = `
<div>
  <img
    alt="The Statue of Liberty's torch heads to new museum"
    class="image"
    src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
  />
</div>
`;